installer/lib/phx_new/gettext_support.ex

```diff
@@ -0,0 +1,44 @@
+defmodule Phx.New.GettextSupport do
+  @moduledoc false
+
+  @doc ~S"""
+  Translates a message using Gettext if `gettext?` is true.
+
+  The role provides context and determines which syntax should be used:
+
+    * `:heex_attr` - Used in a HEEx attribute value.
+    * `:eex` - Used in an EEx template.
+
+  ## Examples
+
+      iex> ~s|<tag attr=#{maybe_gettext("Hello", :heex_attr, true)} />|
+      ~S|<tag attr={gettext("Hello")} />|
+
+      iex> ~s|<tag attr=#{maybe_gettext("Hello", :heex_attr, false)} />|
+      ~S|<tag attr="Hello" />|
+
+      iex> ~s|<tag>#{maybe_gettext("Hello", :eex, true)}</tag>|
+      ~S|<tag><%= gettext("Hello") %></tag>|
+
+      iex> ~s|<tag>#{maybe_gettext("Hello", :eex, false)}</tag>|
+      ~S|<tag>Hello</tag>|
+  """
+  @spec maybe_gettext(binary(), :heex_attr | :eex | :ex, boolean()) :: binary()
+  def maybe_gettext(message, role, gettext?)
+
+  def maybe_gettext(message, :heex_attr, gettext?) do
```

Comment:
Instead of the attribute, please define two functions? `attr_gettext` and `eex_gettext`? Also, we probably already have a module where we define helpers used across templates. Could these functions be moved there? I would try to avoid defining a module only for this purpose.

Reply:
Hi José, thank you for the review!

I could not find a common helper module that both the installer and
Phoenix Mix tasks can access. My understanding from getting familiar with the
code was that dependencies are kept to a minimum, which I appreciate, and one
example of that being that there are two copies of `core_components.ex` kept in
sync by a test that ensures they match byte-for-byte.

A separate Gettext helper file/module seemed like the best option, as that can
be easily kept in sync with the same testing approach of reading from disk and
comparing the two copies.

If you have a different suggestion or know of a common module we could use,
please let me know and I'll be happy to update the patch.

> Instead of the attribute, please define two functions? `attr_gettext` and
`eex_gettext`?

I'm open to making that change. However, I want to ensure that the function
names accurately reflect their behavior. The current approach with
`maybe_gettext` indicates that the output may or may not contain `gettext`
calls. If we use names like `attr_gettext` and `eex_gettext`, it might suggest
that the output always contains `gettext` calls. Could you please clarify what
function signatures you had in mind, I think I might have misunderstood your
suggestion.

Once again, thank you for your feedback!

Reply:
To be clear, I didn't meant a common module for both phoenix and phx_new, but rather a common module within each of those packages instead of a new one in both.

I think `maybe_attr_gettext` and `maybe_eex_gettext` are also fine as function names. :)

Reply:
> To be clear, I didn't meant a common module for both phoenix and phx_new, but rather a common module within each of those packages instead of a new one in both.

Understood. There will be two copies.

On the installer side, an obvious choice would be `installer/lib/phx_new/generator.ex`, because that is where we make the helper available for the template. The other files in `phx_new` are themed around the code they generate.

For Phoenix, we could go with `lib/mix/phoenix.ex`.

In that case, how would you suggest ensuring the helpers stay in sync? I believe the `File.read!` approach would not work as smoothly.

Reply:
I think we don't need them fully in sync, they are small functions anyway. The modules you suggested are good. :)

Reply:
Done! I applied the suggested changes and rebased on top of the latest main branch.